[sh] Simple export to TeX - just to copy-paste into courseworks easily

nickkolok opened the pull request:
Просто так удобнее копировать в курсачи!

nickkolok committed:
[sh] Simple export to TeX - just to copy-paste into courseworks easily

SugarHedgehog commented:
А что по картинкам? Их либо стоит вырезать из экспорта, либо экспортировать  закомментированную строку строку `\includegraphics{}`

nickkolok commented:
Пихать картинки непосредственно в файл несколько проблематично:
https://tex.stackexchange.com/questions/208819/embedding-images-in-tex-file-as-base64-strings
Хотя и возможно:
https://github.com/zerotoc/pdfinlimg

nickkolok committed:
[lib] roughHTML2LaTeX(str)

nickkolok committed:
[sh] use roughHTML2LaTeX(str) in the debug shell

nickkolok committed:
[sh] use roughHTML2LaTeX(str) in the print shell

nickkolok commented:
@SugarHedgehog Вроде как поправлено

SugarHedgehog commented:
Добавить реакцию на нажатие кнопки - появление временного текста "Задание скопировано" (или "Задача в буфере обмена")
Сейчас вообще непонятно отработала ли кнопка

SugarHedgehog commented:
```
Текст задания:

%<canvas style="float:left; margin-right:1em;"\space width="400"\space height="400"\space id="canvas941086864"\space data-nonce="0.7836761135075575">
%</canvas>
%includegraphics{}
В треугольнике $QME$ угол $E$ равен $120^\circ$, $QM =\sqrt{1200}$. Найдите радиус окружности, описанной около этого треугольника.

Ответ:

20

```

Почему строка canvas... ./canvas до сих пор в экспорте?

nickkolok committed:
[sh][css][ref] #gotov -> #readiness-message

nickkolok committed:
[css][ref] Move #readiness-message to main.css - it'll be used more widely

nickkolok committed:
[sh] When replacing canvas with an image, put a newline after it - for better distinction from the following text

nickkolok committed:
[sh] Remove very old rudiment

nickkolok committed:
[lib] roughHTML2LaTeX() - comment the images out

nickkolok committed:
[sh] Convert canvases into images during export to TeX

Now, when copied into TeX, the images are still not ready to use;
however, they can be restored from their base64 codes

nickkolok committed:
[sh] readiness-message in the debug shell - just a stub

nickkolok committed:
[sh] Debug mode - display the message about successful copying

nickkolok commented:
Из новой пачки первый коммит можно перетащить черри-пиком в основную ветку - он вообще попутный. То же относится ко второму или второму вместе с третьим - там предварительный рефакторинг.

Вроде все замечания учтены, всё работает!
